extension/chrome/content/download.js

```diff
@@ -83,18 +87,40 @@ Downloader = {
 
       // If target doesn't exist, create it
       if (!this.target.exists()) {
-        this.target.create(0x00, 0644);
+        this.target.create(Ci.nsIFile.NORMAL_FILE_TYPE, parseInt("0666", 8));
       }
 
-      this._persist = Cc["@mozilla.org/embedding/browser/nsWebBrowserPersist;1"].
-                      createInstance(Ci.nsIWebBrowserPersist);
-      this._persist.persistFlags = Ci.nsIWebBrowserPersist.PERSIST_FLAGS_REPLACE_EXISTING_FILES |
-                                   Ci.nsIWebBrowserPersist.PERSIST_FLAGS_FROM_CACHE |
-                                   Ci.nsIWebBrowserPersist.PERSIST_FLAGS_CLEANUP_ON_FAILURE;
-
-      // Save URL as target
-      this._persist.progressListener = new DownloaderListener();
-      this._persist.saveURI(uri, null, null, null, null, this.target, null);
+      Downloads.createDownload({'source': uri, 'target': this.target}).then((aDownload) => {
+        var maxMB = 0;
+        var currentMB = 0;
+
+        // The download final size and progress percentage is unknown.
+        Downloader.progressMeter.mode =  "undetermined";
```

Comment:
Based on which promise you set this to undetermined? Will it be automatically set to determined when you update the value property? If that is the case you should define this default in the appropriate XUL file.